### PR TITLE
Update the storage documentation

### PR DESCRIPTION
**Pull request by metalmatze**
As another follow-up, I think we need to update the `by-id` meta store reference. I could only find `by-key` and this is what I remember we're doing now.

Eventually, we should document how we ingest Arrow and through compaction go about storing Parquet files. Not as important for now, but down the road, once persistence is a default, we should add it too.

### CODE REVIEW

File: docs/storage.md [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
 
-Requests to store profiles are either coming from Parca's scraper, which collects profiles by scraping a targets' HTTP endpoint or coming from a process that is writing into Parca directly, like the [Parca agent](parca-agent).
+Requests to store profiles are either coming from Parca's scraper, which collects profiles by scraping a target's HTTP endpoint or coming from a process that is writing into Parca directly, like the [Parca agent](https://github.com/parca-dev/parca/issues/parca-agent).
 
-Upon receiving such a request, Parca uses [pprof's upstream parser](https://pkg.go.dev/github.com/google/pprof/profile#Parse) to parse and validate these profiles. This ensures that the profiles are valid pprof profiles but also that Parca is compatible with pprof.
+Upon receiving a request, Parca unmarshalls the request into the upstream [pprof protobuf](https://github.com/parca-dev/parca/blob/main/proto/google/pprof/profile.proto) definition. This ensures that the profiles are valid pprof profiles but also that Parca is compatible with pprof.
```
Comment (asubiotto) [inline docs/storage.md:11]: nit: I believe unmarshals is spelt with just a single 'l'

File: docs/storage.md [hunk 2]
```diff
@@ -49,31 +49,33 @@ Profiling data at a high level can be described as stacktraces with a value atta
...
+| 6 x mypod1 | 6 x mynode1 | 3 x t1    | main;func1;func2 | 2     |
+|            |             |           | main;func1;func3 | 23    |
+|            |             |           | main;func1;func4 | 3     |
+|            |             | 2 x t2    | main;func1;func2 | 10    |
+|            |             |           | main;func1;func3 | 12    |
+|            |             | t3        | main;func1;func3 | 234   |
+
+Sorting by the dynamic columns and immediately followed by the timestamp column has been shown in benchmarks to be an effective way of ignoring data that is outside of the requested time range. In our example above, we might only be interested in the `cpu{pod="mypod1",node="mynode1"}` from `t2` to `t3`. Upon iterating over the data, FrostDB ignores all the data below `t2`.
 
 Read more on how arcticDB and Parca were originally created in the announcement blog post: https://www.polarsignals.com/blog/posts/2022/05/04/introducing-arcticdb/
```
Comment (asubiotto) [inline docs/storage.md:81]: nit: arcticDB still referenced here

File: docs/storage.md [hunk 3]
```diff
@@ -1,43 +1,43 @@
 # Storage
 
-Parca has two distinct storage components: metastore and samplestore. The metastore stores metadata such as function names, line numbers, file names, and so on. The samplestore stores the profiling data.
+Parca has two distinct storage components: meta store and sample store. The meta store stores metadata such as function names, line numbers, file names, and so on. The sample store stores the profiling data.
 
-## Metastore
+## Meta Store
```
Comment (kakkoyun) [inline docs/storage.md:5]: FYI we might have used `metastore` as a single word everywhere in the docs and other repositories.

Comment (metalmatze) [reply docs/storage.md:5]: I was wondering if there is a convention we follow.
Having the metastore next to the sample store seems wrong and thus I went with meta store. hmm.

File: docs/storage.md (hunk 2 above)
Comment (metalmatze) [reply docs/storage.md:81]: funny, how did I miss that? 😄

File: docs/storage.md (hunk 3 above)
Comment (kakkoyun) [reply docs/storage.md:5]: I don't mind which one to use, as long as we're consistent :)

**Review by brancz — APPROVED** [review]

**Pull request merged**